app/controllers/api.rb

```diff
@@ -1,2 +1,18 @@
 module Api
+  class RootController < ApplicationController
+    skip_before_action :validate_primary_collection_id
+    skip_before_action :validate_request
+
+    def head_request_exists_in_get_routes?
+      routes = Rails.application.routes.recognize_path(request.path, :method => :get)
+      routes.extract!(:controller, :action) != {:controller => "api/root", :action => "invalid_url_error"}
+    end
+
+    def invalid_url_error
+      return head :ok if request.head? && head_request_exists_in_get_routes?
```

Comment:
Should this ever render `:ok`?  We only get here is the url is invalid, right?

Reply:
@lpichler would you be able to provide any followup to @bdunne?

Reply:
I removed logic for `HEAD` request because it need to solved differently - maybe by adding head routes to `routes.rb`

Reply:
I removed logic for HEAD and we need to investigate whether there is better way how to handle HEAD request.